src/verifier.hpp

```
@@ -0,0 +1,21 @@
+#include "ir/program.hpp"
+#include "result.hpp"
+
+namespace prevail {
+
+AnalysisResult analyze(const Program& prog);
+AnalysisResult analyze(const Program& prog, const StringInvariant& entry_invariant);
+inline bool verify(const Program& prog) { return !analyze(prog).failed; }
+
+void ebpf_verifier_clear_thread_local_state();
+
+struct ThreadLocalGuard {
```

Clang (14.0.6)

[warning] class 'ThreadLocalGuard' defines a non-default destructor but does not define a copy constructor, a copy assignment operator, a move constructor or a move assignment operator

(cppcoreguidelines-special-member-functions)